content/compose/environment-variables/variable-interpolation.md

```diff
@@ -211,3 +212,42 @@ If an environment variable is not set, Compose substitutes with an empty string.
 > **Note**
 >
 > `postgres:` is not a valid image reference. Docker expects either a reference without a tag, like `postgres` which defaults to the latest image, or with a tag such as `postgres:15`.
+
+
+#### local `.env` file _vs_ <project directory> `.env` file
+
+An `.env` file can also be used to declare [`COMPOSE_*` variables](envvars.md) used to control Compose behavior and files to be loaded. When executed without an
+explicit `--env-file` flag, Compose will search for a `.env` file in working directory (PWD) and load values both for self-configuration and interpolation. If
+those values define `COMPOSE_FILE` variable which results into project directory being set to another folder, then the latter is also used to load a second
+`.env` file, which will have lower precedence. 
+
+This mechanism makes it possible to invoke an existing compose project with a custom set of variables as overrides, without the need to pass environment 
+variables by the command line.
+
+```console
+$ cat .env
+COMPOSE_FILE=../compose.yaml
+POSTGRES_VERSION=9.3
+
+$ cat ../compose.yaml 
+services:
+  db:
+    image: "postgres:${POSTGRES_VERSION}"
+$ cat ../.env
+POSTGRES_VERSION=9.2
+
+$ docker compose config
+services:
+  db:
+    image: "postgres:9.3"
+```
+
+
+#### diagnose interpolatin issue
+
+Variable can be declared by multiple sources to be used by interpolation. Precedence applies as:
+1. variables from shell environment
+2. if `--env-file` is not set, variables set by an `.env` file in local working directory (PWD)
+3. variables from file set by `--env-file` or an `.env` file in project directory
```

Comment:
What about `COMPOSE_FILE` and the explicitly envs inside the compose config?

Reply:
AFAICT this is the purpose of the "local `.env` file _vs_ <project directory> `.env` file" section
or do you have something in mind ?

Reply:
Just thought we could add all the possibilities in this list